File: app-ios/Packages/module-ai/Sources/ModuleAI/Application/Models/AIChatIntentResolver.swift
//
//  AIChatIntentResolver.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/1.
//

import Foundation

enum AIChatIntentResolver {
    static func resolve(arguments: FunctionArguments) -> AIChatIntentType? {
        switch arguments {
        case .timeRange(let timeRange):
            if timeRange.startDate == nil {
                return .time
            }
        case .performanceType:
            return .index
        default:
            break
        }
        
        return nil
    }
}

File: app-android/feature/ai-chat/src/main/java/com/aidatainsight/android/feature/aichat/application/AIChatIntentResolver.kt
package com.aidatainsight.android.feature.aichat.application

import com.aidatainsight.android.core.model.contract.AIChatIntentType
import com.aidatainsight.android.core.model.contract.FunctionArguments

object AIChatIntentResolver {
    fun resolve(arguments: FunctionArguments): AIChatIntentType? {
        return when (arguments) {
            is FunctionArguments.TimeRange -> {
                if (arguments.value.startDate.isNullOrBlank()) AIChatIntentType.Time else null
            }
            is FunctionArguments.PerformanceType -> AIChatIntentType.Index
            else -> null
        }
    }
}

